src/tests/JIT/HardwareIntrinsics/Arm/Shared/VectorLookupExtension_2Test.template

```diff
@@ -0,0 +1,400 @@
+
+            // Validates passing a local works, using Unsafe.Read
+            test.RunLclVarScenario_UnsafeRead();
+
+            // Validates passing the field of a local class works
+            test.RunClassLclFldScenario();
+
+            // Validates passing an instance member of a class works
+            test.RunClassFldScenario();
+
+            // Validates passing the field of a local struct works
+            test.RunStructLclFldScenario();
+
+            // Validates passing an instance member of a struct works
+            test.RunStructFldScenario();
+
+            if (!test.Succeeded)
+            {
+                throw new Exception("One or more scenarios did not complete as expected.");
+            }
+        }
+    }
+
+    public sealed unsafe class VectorLookupExtension_2Test__{Method}{RetBaseType}
+    {
+        private struct DataTable
+        {
+            private byte[] inArray0;
+            private byte[] inArray1;
+            private byte[] inArray2;
+            private byte[] inArray3;
+            private byte[] outArray;
+
+            private GCHandle inHandle0;
+            private GCHandle inHandle1;
+            private GCHandle inHandle2;
+            private GCHandle inHandle3;
+            private GCHandle outHandle;
+
+            private ulong alignment;
+
+            public DataTable({Op2BaseType}[] inArray0, {Op1BaseType}[] inArray1, {Op1BaseType}[] inArray2, {Op2BaseType}[] inArray3, {RetBaseType}[] outArray, int alignment)
+            {
+                int sizeOfinArray0 = inArray0.Length * Unsafe.SizeOf<{Op2BaseType}>();
+                int sizeOfinArray1 = inArray1.Length * Unsafe.SizeOf<{Op1BaseType}>();
+                int sizeOfinArray2 = inArray2.Length * Unsafe.SizeOf<{Op1BaseType}>();
+                int sizeOfinArray3 = inArray3.Length * Unsafe.SizeOf<{Op2BaseType}>();
+                int sizeOfoutArray = outArray.Length * Unsafe.SizeOf<{RetBaseType}>();
+                if ((alignment != 32 && alignment != 16 && alignment != 8) || (alignment * 2) < sizeOfinArray0 || (alignment * 2) < sizeOfinArray1 || (alignment * 2) < sizeOfinArray2 || (alignment * 2) < sizeOfinArray3 || (alignment * 2) < sizeOfoutArray)
+                {
+                    throw new ArgumentException("Invalid value of alignment");
+                }
+
+                this.inArray0 = new byte[alignment * 2];
+                this.inArray1 = new byte[alignment * 2];
+                this.inArray2 = new byte[alignment * 2];
+                this.inArray3 = new byte[alignment * 2];
+
+                this.inHandle0 = GCHandle.Alloc(this.inArray0, GCHandleType.Pinned);
+                this.inHandle1 = GCHandle.Alloc(this.inArray1, GCHandleType.Pinned);
+                this.inHandle2 = GCHandle.Alloc(this.inArray2, GCHandleType.Pinned);
+                this.inHandle3 = GCHandle.Alloc(this.inArray3, GCHandleType.Pinned);
+                this.outHandle = GCHandle.Alloc(this.outArray, GCHandleType.Pinned);
+
```

Comment:
This accesses this.outArray which is not set, so its null.

Reply:
Good catch. Fixed. I will check why CI didn't complain about it.

Reply:
The CI was passing because Arm64 test cases were not running. https://github.com/dotnet/runtime/pull/83948 will fix that.